packages/sanity/src/structure/panes/document/DocumentOperationResults.tsx

```diff
@@ -11,13 +11,25 @@ const IGNORE_OPS = ['patch', 'commit']
 
 export const DocumentOperationResults = memo(function DocumentOperationResults() {
   const {push: pushToast} = useToast()
-  const {documentId, documentType} = useDocumentPane()
-  const {title} = useDocumentTitle()
+  const {documentId, documentType, value: documentPaneValue} = useDocumentPane()
+  const documentTitleInfo = useDocumentTitle()
+  let title = documentTitleInfo.title
+  const titleError = documentTitleInfo.error
   const event: any = useDocumentOperationEvent(documentId, documentType)
   const prevEvent = useRef(event)
   const paneRouter = usePaneRouter()
   const {t} = useTranslation(structureLocaleNamespace)
 
+  if (
```

Comment:
this may be more react-esque if it was calculated. e.g.

```
  const title = useMemo(() => {
    // If title isn't set from document preview, use the title from the document pane value
    if (
      !documentTitleInfo.title &&
      !titleError &&
      !IGNORE_OPS.includes(event?.op) &&
      typeof documentPaneValue.title === 'string' &&
      event?.type === 'success'
    ) {
      return documentPaneValue.title
    }
    return documentTitleInfo.title
  }, [documentTitleInfo.title, titleError, event, documentPaneValue.title])
```

Reply:
👍 Thanks for that suggestion, I was going back and forth on the decision to memoize it.

Reply:
its not too important to memoize it since the value isn't an object reference and is cheap to calculate but it kinda feels better to useMemo than to have an IIFE e.g. `const title = (() => {})()` do the same thing. mutating the variable is the thing to avoid in react renders

Reply:
Yeah, I like the memo approach better. I went ahead and made that change.